connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectAnalyzeHandler.scala

```diff
@@ -201,7 +201,9 @@ private[connect] class SparkConnectAnalyzeHandler(
       case other => throw InvalidPlanInput(s"Unknown Analyze Method $other!")
     }
 
-    builder.setSessionId(request.getSessionId)
+    builder
+      .setSessionId(request.getSessionId)
+      .setServerSideSessionId(session.sessionUUID)
```

Comment:
Out of overabundance of caution, I would instead of doing sessionHolder.sessionUUID add a method to SessionHolder

```
def serverSessionId = {
  // The server side session id must be different from the client generated one
  assert(session.sessionUUID) != sessionId
  session.sessionUUID
}
```

to prevent that someone in the future decides to sync these session.sessionUUID with the connect session_id, which would make this idempotency check not effective.

Reply:
Done